llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -167,20 +216,143 @@ std::pair<StringRef, unsigned> getSpecConstMetadata(Instruction *I) {
   return std::make_pair(MDSym->getString(), ID);
 }
 
-static Value *getDefaultCPPValue(Type *T) {
-  if (T->isIntegerTy())
-    return Constant::getIntegerValue(T, APInt(T->getScalarSizeInBits(), 0));
-  if (T->isFloatingPointTy())
-    return ConstantFP::get(T, 0);
-  llvm_unreachable("unsupported spec const type");
-  return nullptr;
+void collectCompositeElementsInfoRecursive(
+    const Type *Ty, unsigned &Index, unsigned &Offset,
+    std::vector<CompositeSpecConstDescriptor> &Result) {
+  if (auto *ArrTy = dyn_cast<ArrayType>(Ty)) {
+    for (size_t I = 0; I < ArrTy->getNumElements(); ++I) {
+      // TODO: this is a spot for potential optimization: for arrays we could
```

Comment:
```suggestion
      // TODO: this is a spot for potential optimization: for arrays & vectors we could
```

Reply:
Fixed in 4c9ce329cecc0563d51fd71aa56696fa2ff40dba